### ISSUE TITLE
Document Supported Database Versions for plugins

### ISSUE THREAD

**Issue by kript** [irods/irods#3819]
### iRODS Version, OS and Version

4.1.10 (but also interested in 4.2.x)

### What did you try to do?

Find documentation as to which database plugin supports which version of the RDBMA, in order to support internal requirement of upgrading Oracle database.

### Expected behavior

A matrix of Database plugins with supported versions of each RDMA against each version.

### Observed behavior

No list found

### Steps to reproduce

View
https://docs.irods.org/4.2.2/plugins/pluggable_database/
https://docs.irods.org/4.1.10/manual/architecture/#pluggable-database

Fail to see any mention of database versions. :-)

**Comment by trel** [comment]
Fair enough.  We've tried to keep the SQL as simple and generic as possible - in order to be as compatible as possible across database management systems and versions.

The oldest system we have in CI is CentOS6 which provides via package manager:
 - Oracle Express Edition 11g 11.2.0
 - MySQL 5.5.56
 - PostgreSQL 8.4.20

This is true for both iRODS 4.1.x and up.

**Comment by kript** [comment]
Thanks for that. Do you test against newer versions as well, i.e., is there any expectation that later than 11.2 will work, for Oracle, for e.g. (we're being requested to migrate to 12.1, for e.g.)?

**Comment by trel** [comment]
We are testing against the versions made available by the OS distribution.  We don't have the capacity at this time to test more than what the distributions test/package/maintain themselves.

That said, we have no reason to think that a newer version of a database will not work as well as an older, tested version.  Our SQL has not changed, and the SQL we use has been designed to be as compatible as possible (we have not optimized each database plugin to use database-specific features except where necessary/incompatible otherwise).

I expect Oracle 12.1 will work without incident - but we have no tests within the Consortium at this time testing against 12.1.

**Comment by kript** [comment]
While installing 4.1.10 with the Oracle plugin on Red Hat, I noticed that it had been built with Oracle Instant Client 11.1, which means thats what it will expect to be installed, as it looks for libraries that are version named. 

As this is slightly different than the Ubuntu support for 11.2 that we're also using quite happily, I thought its worth raising, although I'm not ure this is the right issue (happy to create a new one if needed)?

**Comment by trel** [comment]
Hm, interesting.  I didn't realize they weren't using the same version themselves (CentOS6 and CentOS7)...   I think this is a fine place to note this.

**Comment by kript** [comment]
Given the Oracle client is now at 19.04, is it time to bump the tested client? Happy to raise a separate issue for this, especially as we are looking to move to latest LTS of Instant Client (which may not be 19.04)...

**Comment by trel** [comment]
We'll probably need some additional hand-holding to get our CI smart enough to handle these multiple versions of things.   I think Sanger is the only Consortium Member using Oracle at this time.

**Comment by kript** [comment]
😿 Has everyone else moved to other databases then?   There used to be more of us...

**Comment by kript** [comment]
Also, let us know what help you need and I'll see if I can get assistance from Our Fine Team of DBA's

**Comment by kript** [comment]
>While installing 4.1.10 with the Oracle plugin on Red Hat 7, I noticed that it had been built with >Oracle Instant Client 11.1, which means thats what it will expect to be installed, as it looks for >libraries that are version named.

FWIW it looked like the Ubuntu plugin has also been built this way;

Ubuntu 18.04 server with 4.1.12;
```
Configuration problem:
    Communications with the database failed.
        /var/lib/irods/iRODS/server/test/bin/test_cll: error while loading shared libraries: libclntsh.so.11.1: cannot open shared object file: No such file or directory
```

Oracle versions of plugin and Instant Client;
```
# dpkg -l | grep oracle
ii  irods-database-plugin-oracle     1.12                                        amd64        iRODS Oracle Database Plugin
ii  oracle-instantclient12.1-basic   12.1.0.2.0-2                                amd64        Oracle Instant Client Basic package
ii  oracle-instantclient12.1-devel   12.1.0.2.0-2                                amd64        Development headers for Instant Client.
ii  oracle-instantclient12.1-sqlplus 12.1.0.2.0-2                                amd64        SQL*Plus for Instant Client.
```

`ldd` is the smoking gun
```
# ldd /var/lib/irods/iRODS/server/test/bin/test_cll
        linux-vdso.so.1 (0x00007ffe06943000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa77544d000)
        libclntsh.so.11.1 => not found
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa775249000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa77502a000)
        libssl.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007fa774dc2000)
        libcrypto.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007fa77497f000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fa7745f6000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa774258000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fa774040000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa773c4f000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fa775655000)
```
